### ISSUE TITLE
Hopper Duplication with Premium Plugin UpgradeableHoppers

### ISSUE THREAD

**Issue by HellaMixel** [GC-spigot/AdvancedEnchantments#3091]
### Describe the bug

This is a Plugin issue with version AdvancedEnchantments-9.0.0b29.jar and with the premium plugin UpgradeableHoppers

When blacklisting Hoppers for the enchant Trench it will not destroy a hopper, but if a Chunk Hopper is in it's place it does not recognize it as a Hopper and will duplicate. 

### How to reproduce

Steps to recreate:

Place a Chunkhopper on say a ceiling, place a block under the Chunk Hopper, and use Trench enchant. 

What will happen is the Trench will activate and replace the Chunk Hopper with a regular Hopper, and then give you a Chunk Hopper. Therefore, duplicating a Chunk Hopper for a regular Hopper

### Screenshots / Videos

https://imgur.com/a/X8NcIbE

### "/ae plinfo" link

https://paste.md-5.net/uvacujuvoh

### Server Log

https://mcpaste.io/1b8540b87b816004

**Comment by Awpmeister2170** [comment]
Hello, with these type of issues its hard for us to look into as its a Premium plugin and we dont own them to test potential fixes against it 

Was this working in previous versions?

**Comment by HellaMixel** [comment]
> Hello, with these type of issues its hard for us to look into as its a Premium plugin and we dont own them to test potential fixes against it
> 
> Was this working in previous versions?

I’m not sure, this was a bug just brought to my attention yesterday

**Comment by HellaMixel** [comment]
I was on the discord and was told to come here for the issue, not sure where or who to report this to, because if I go to UpgradableHoppers they’d just say the same thing?

**Comment by Awpmeister2170** [comment]
It may be worth contacting them to see if they would be happy for you or themselves  to provide us a copy of the plugin so we can work on a fix for this.

Unfortunately due to neither me or any of our developers owning this plugin it’s difficult for us to come to a solution for it

**Comment by HellaMixel** [comment]
> It may be worth contacting them to see if they would be happy for you or themselves to provide us a copy of the plugin so we can work on a fix for this.
> 
> Unfortunately due to neither me or any of our developers owning this plugin it’s difficult for us to come to a solution for it

I will reach out to them. Question, do you need the actual plugin itself or do you just need the server to test it on? I can grant access to /OP on the server, but if it was for console access I can work something out. Or is that still technically going against their terms of distribution? As long as you’re not downloading the plugin? 

Just want to try and get this issue resolved before too many people find out about it.

**Comment by GC-spigot** [comment]
Yes, we call it here and there for enchants which break large areas to test if blocks are breakable. I’ll see what we can do to resolve it. On 2022-11-07, at 09:07 AM, Angeschossen ***@***.***> wrote:﻿
Added GC to the buyer's list.
To me, it sounds like some event, like BlockBreakEvent is called, without actually removing the block?
